sql/core/src/test/scala/org/apache/spark/sql/execution/metric/SQLMetricsSuite.scala

```diff
@@ -268,6 +275,19 @@ class SQLMetricsSuite extends SharedSparkSession with SQLMetricsTestUtils {
           "remote blocks read" -> 0L,
           "shuffle records written" -> 2L))))
       )
+
+      val df1 = spark.sql(query)
+      testSparkPlanMetrics(df1, 1, Map(
+        1L -> (("SortMergeJoin", Map(
+          // It's 4 because we only read 3 rows in the first partition and 1 row in the second one
+          "number of output rows" -> 4L))),
+        4L -> (("Exchange", Map(
+          "records read" -> 4L,
+          "local blocks read" -> 2L,
+          "remote blocks read" -> 0L,
+          "shuffle records written" -> 2L)))),
+        true
```

Comment:
is this just a repeat of the above block of code with an extra `true` argument?  If so, better to refactor like

```scala
Seq(true, false).foreach { wholeStageCodeGen =>
 testSparkPlanMetrics(..., wholeStageCodeGen)
}
```

same throughout all the changes.